lib/serializer.js

```diff
@@ -323,8 +323,11 @@ class Serializer {
         }
       }
       ```
+
+      You can also define `embed` as a function so it can be determined dynamically.
     */
     this.embed = this.embed || undefined; // this is just here so I can add the doc comment. Better way?
+    this.embedFn = isFunction(this.embed) ? this.embed : () => !!this.embed;
```

Comment:
This doesn't actually work, right?  An ES6 derived serializer will not be able to set `this.embed` before this code runs, effectively making `this.embedFn = () => true;`

Reply:
I'm not sure I quite follow, @navels.  What do you mean by "An ES6 derived serializer"?

Reply:
Aha, this is only a problem when defining the `embed` function as a class property, which is an ES2022 thing.  E.g.,

```
class Serializer {
  constructor() {
    console.log(`typeof this.embed: ${typeof this.embed}`);
  }
}

class FooSerializer extends Serializer {
  /* classical class method declaration */
  embed(key) {
    return false;
  }
}

class BarSerializer extends Serializer {
  /* ES2022 class property form */
  embed = (key) => false;
}

new FooSerializer();
/* console output: "typeof this.embed: function" */

new BarSerializer();
/* console output: "typeof this.embed: undefined" */
```
the difference being in the latter case `this.embed` doesn't exist until after the constructor has run.

So we can also work around this by not using the ES2022 syntax, but still. Kinda interesting potential class of bugs that new syntax is going to expose.